draft-ietf-secevent-http-push.xml

```diff
@@ -384,41 +374,41 @@ Content-Type: application/json
 
             <section title="Denial of Service">
                 <t>
-                    The SET Receiver may be vulnerable to a denial-of-service attack where a
+                    The SET Recipient may be vulnerable to a denial-of-service attack where a
                     malicious party makes a high volume of requests containing invalid SETs,
                     causing the endpoint to expend significant resources on cryptographic
                     operations that are bound to fail. This may be mitigated by authenticating
                     SET Transmitters with a mechanism with low runtime overhead, such as mutual
-                    TLS or statically assigned bearer tokens.
+                    TLS.
                 </t>
             </section>
 
             <section title="Authenticating Persisted SETs">
                 <t>
-                    At the time of receipt, the SET Receiver can rely upon transport layer
+                    At the time of receipt, the SET Recipient can rely upon transport layer
                     mechanisms, HTTP authentication methods, and/or other context from the
                     transmission request to authenticate the SET Transmitter and validate the
                     authenticity of the SET.  However, this context is typically unavailable to
-                    systems that the SET Receiver forwards the SET onto, or to systems that
-                    retrieve the SET from storage.  If the SET Receiver requires the ability to
+                    systems that the SET Recipient forwards the SET onto, or to systems that
+                    retrieve the SET from storage.  If the SET Recipient requires the ability to
                     validate SET authenticity outside of the context of the transmission request,
-                    then the SET Transmitter SHOULD sign the SET in accordance with <xref target="RFC7515"/>,
-                    or encrypt it using an authenticated encryption scheme in accordance with <xref target="RFC7516"/>.
+                    then the SET Transmitter SHOULD sign the SET in accordance with <xref target="RFC7515"/>
+                    and optionally also encrypt it in accordance with <xref target="RFC7516"/>.
```

Comment:
I specifically called out the use of "authenticated encryption scheme" as a typical application of asymmetric crypto does not generally guarantee message authenticity, and I think we want to be explicit about recommending a crypto operation that does that. I'm happy to workshop better language for that.

(aside) It'd be nice of we could point to a JWE/JWS BCP doc for this.

Reply:
Yes - that was intentional.  Using authenticated encryption only proves that you encrypted the message to the recipient using the recipient's encryption public key.  It says nothing about who did the encryption.  Anyone with access to the recipient's encryption public key can do that.

The issuer needs to sign the SET if the recipient is to be able to cryptographically validate the issuer.